widget/list.go

```diff
@@ -93,6 +93,36 @@ func (l *List) MinSize() fyne.Size {
 	return l.BaseWidget.MinSize()
 }
 
+func (l *List) refreshBase() (*listLayout, map[ListItemID]*listItem, fyne.Focusable) {
+	if l.scroller != nil {
+		l.BaseWidget.Refresh()
+		lo := l.scroller.Content.(*fyne.Container).Layout.(*listLayout)
+		visible := lo.visible
+		canvas := fyne.CurrentApp().Driver().CanvasForObject(lo.list)
+		var focused fyne.Focusable
+		if canvas != nil {
+			focused = canvas.Focused()
+		}
+		return lo, visible, focused
+	}
+	return nil, nil, nil
+}
+
+func (l *List) Refresh() {
```

Comment:
By adding this function you are cutting out the existing Refresh work in the renderer (having overridden the BaseWidget.Refresh()).
Just looking at it this may lead to duplications, or could be redundant code in renderer now.

Reply:
I'm not sure I follow. The BaseWidget refresh is being called in refreshBase. I haven't looked much into it yet, but it looks like it handles the renderer refresh.